src/udev/udev-builtin-hwdb.c

```diff
@@ -99,6 +99,18 @@ static int udev_builtin_hwdb_search(sd_device *dev, sd_device *srcdev,
                         last = true;
                 }
 
+                if (streq(dsubsys, "firewire")) {
+                        const char *devname;
+
+                        // The node device has character device while the unit device doesn't.
+                        if (!sd_device_get_devname(d, &devname) && strlen(devname) > 0) {
+                                // The parent of node is 1394 OHCI controller in PCI bus. Let us
+                                // cancel walkthrough.
+                                last = true;
+                                r = -ENOENT;
```

Comment:
It is enough just return here with `return -ENOENT;`, instead of setting `last` and `r`.

Reply:
Yes. Within the single patch, the short return looks good.

However, as I  note in request message, I'm working for additional patchset to use custom modalias-like key for hwdb entry. The last local variable and the following call of udev_builtin_hwdb_lookup() is required.

Reply:
OK.